presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/MergeLimitWithDistinct.java

```diff
@@ -31,20 +31,11 @@
 {
     private static final Capture<AggregationNode> CHILD = newCapture();
 
-    private static final Pattern<LimitNode> PATTERN = limit()
-            .with(source().matching(aggregation().capturedAs(CHILD)
-                    .matching(MergeLimitWithDistinct::isDistinct)));
-
     /**
-     * Whether this node corresponds to a DISTINCT operation in SQL
+     * Apply this rule only if aggregation node corresponds to a DISTINCT operation in SQL
```

Comment:
once you renamed `AggregationNode::producesDistinctRows` method, this comment is no longer needed